homeassistant/components/generic/config_flow.py

```diff
@@ -309,6 +316,40 @@ def register_preview(hass: HomeAssistant) -> None:
     hass.data[DOMAIN][IMAGE_PREVIEWS_ACTIVE] = True
 
 
+async def register_stream_preview(hass: HomeAssistant, config) -> str:
+    """Set up preview for camera stream during config flow."""
+    hass.data.setdefault("camera", {})
+
+    # Need to load the camera prefs early to avoid errors generating the stream
+    # if the user does not already have the stream component loaded.
+    if hass.data.get(DATA_CAMERA_PREFS) is None:
+        prefs = CameraPreferences(hass)
+        await prefs.async_load()
+        hass.data[DATA_CAMERA_PREFS] = prefs
+
+    # Create a camera but don't add it to the hass object.
```

Comment:
Is it a problem that stream may call `async_write_ha_state` on this if its not really added to hass?

Reply:
If it gets called, then I think it will write something to the hass data object. But is stream likely to do that?

Either way I don't see it causing a problem other than a data entry for something that will no longer exist in the future. I have looked at a few ways but think it could be hard to decouple this.

Possibly could do something like 
`self._platform_state == EntityPlatformState.REMOVED`
to block the write but that is a bit of a hack. Maybe there is a better way of doing this?

Possibly subclass it and modify `async_write_ha_state`? Maybe there is a general case for this type of throw away temporary entity to handle previews in config/option flows?

Reply:
@allenporter I've done some experimentation, and haven't seen the stream writing ` _async_write_ha_state` so far in my logs. I couldn't see a mechanism where this would happen other than one being added via `set_update_callback()`, and nothing seems to do that.

Please let me know if I'm missing something! 

I think this is working ok now from the core side.